src/routes/blog/post/user-role-guests-missing-scope-account/+page.markdoc

```diff
@@ -0,0 +1,118 @@
+
+There are a handful of reasons you may be receiving this error, which we'll explore in detail below.
+
+# No active session
+
+The most common cause of this error is attempting to access user account information when there's no active session.
+
+If you're not logged in yet and you try to get information about the user with something like `account.get()`, you'll see this error.
+
+```jsx
+// This will fail if no user is logged in
+try {
+  const user = await account.get();
+  console.log(user);
+} catch (error) {
+  console.error(error); // Error: User (role: guests) missing scope (account)
+}
+```
+
+This error is actually expected and useful in certain scenarios, such as when your app first starts and you need to check if a user is logged in. You can catch this error to determine that there's no active session and redirect users to a login page.
+
+However, if you're getting this error when you actually need an authenticated session, simply create a session before calling `account.get()`:
+
+```jsx
+// First create a session
+await account.createEmailSession('user@example.com', 'password');
+
+// Now this will work
+const user = await account.get();
+console.log(user);
+```
+
+# OAuth2 sessions in web applications
+
+A more complex scenario occurs when using OAuth2 authentication in web applications. When you use `account.createOAuth2Session()` in a web app, a cookie is set for the session.
+
+If your app and Appwrite endpoint are on different domains (for example, your app is on `localhost:3000` and you're using `https://cloud.appwrite.io/v1` as your Appwrite endpoint), the cookie may not be included in the `account.get()` request because the cookie is considered a 3rd party cookie.
+
+There are two approaches to solve this issue:
+
+## Option 1: Browser settings and custom domains
+
+For local development:
+
+- Allow 3rd party cookies in your browser settings
+- Some browsers have specific development settings to allow cross-domain cookies during local development
+
+For production:
+
+- [Configure a custom domain](https://appwrite.io/docs/advanced/platform/custom-domains) for your Appwrite project so that cookies are sent from the same domain as your application
+- This ensures that session cookies are properly sent with each request
+
+## Option 2: Use OAuth2 tokens
+
+Another approach is to use Appwrite's OAuth2 token-based authentication, which bypasses third-party cookie restrictions entirely. Instead of `createOAuth2Session()`, you would use `createOAuth2Token()` which implements a more privacy-friendly flow that works in all browsers.
+
+For a detailed implementation guide, check out this article on [Fixing OAuth2 authentication issues in Appwrite Cloud](https://appwrite.io/blog/post/fixing-oauth2-issues-in-appwrite-cloud?doFollow=true), which covers this approach in depth.
+
+Keep in mind that non-public files still require session cookies, so stick with Option 1 if your app needs to handle such files.
```

Comment:
> so stick with Option 1 if your app needs to handle such files.

You can still go with option 2, but you'll need to set up custom domains so the cookie is a first party cookie.

Reply:
Updated